File: Domain/Entities/Curriculum/SyllabusElements/Science.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities.Curriculum.SyllabusElements;

/// <summary>
/// A collection of courses related with particular science.
/// </summary>
public class Science
{
    public int Id { get; set; }
    public string Name { get; set; }
    
    private readonly List<Course> courses = new();

    public Science(string name)
    { 
        Name = name;
    }

    public void AddNewCourse(Course course)
    {
        courses.Add(course);
    }

    public void RemoveCourse(Course course) 
    { 
        courses.Remove(course);
    }
}

File: Domain/Entities/Curriculum/SyllabusElements/Science.java


import java.util.ArrayList;
import java.util.List;

public class Science {
    private int id;
    private String name;
    private List<Course> courses = new ArrayList<>();

    public Science(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Course> getCourses() {
        return courses;
    }

    public void addNewCourse(Course course) {
        courses.add(course);
    }

    public void removeCourse(Course course) {
        courses.remove(course);
    }
}

